### ISSUE TITLE
Unsupported seccfg structure

### ISSUE THREAD

**Issue by brunothedev** [bkerler/mtkclient#634]
it seems that mtkclient reads and writes the whole partition, this can be a problem, for example: on ```mtk da seccfg unlock```, it has a ```0x4D4D4D4D```at the start and a ```0x45454545```at the end, but because it reads the whole partition, it reads the partition header(not the seccfg one) and at the end it reads the whole empty bytes and not the end header.

CPU: MT6735M

**Comment by brunothedev** [comment]
please @bkerler see this, this is very important and breaks most of MTKClient features

**Comment by bkerler** [comment]
Can you please provide an example read ?

**Comment by brunothedev** [comment]
[seccfg.bin](https://drive.google.com/file/d/1AZ--fq7hCWO1a-VDTmbDhNpUIYMD48qI/view?usp=share_link)

As you can see, it has a ```0x4D4D4D4D``` and a ```0x45454545```, but because i starts with ```AND_SECCFG_v``` and ends with empty bytes, MTKClient doesnt identify it

**Comment by brunothedev** [comment]
@bkerler Update?

**Comment by bkerler** [comment]
No update yet. But will work next on it.

**Comment by brunothedev** [comment]
also, may i mnetion that mtkclient identifies my device as "MT6737M/MT6735G()"

**Comment by bkerler** [comment]
@brunothedev This seccfg structure isn't supported yet. Can you send me the preloader and the device vendor/model ?

**Comment by brunothedev** [comment]
@bkerler: [preloader_muse6735m_e1lte_l.bin](https://drive.google.com/file/d/14ovrLtjjRcNqP5F0RagxuqXA4WJe8vFp/view?usp=share_link)

My device is a LG K4 lg-k130f (me1ds)
Chipset: MT6735M

**Comment by brunothedev** [comment]
@bkerler I would like to say, that i think that i am the first person to ever try to install a custom os on it, there is no custom roms, nothing

**Comment by brunothedev** [comment]
@bkerler: I attempted to unlock the bootloader again, here is what happens:

```
..Port - Device detected :)
Preloader - 	CPU:			MT6737M/MT6735G()
Preloader - 	HW version:		0x0
Preloader - 	WDT:			0x10212000
Preloader - 	Uart:			0x11002000
Preloader - 	Brom payload addr:	0x100a00
Preloader - 	DA payload addr:	0x201000
Preloader - 	CQ_DMA addr:		0x10217c00
Preloader - 	Var1:			0x28
Preloader - Disabling Watchdog...
Preloader - HW code:			0x335
Preloader - Target config:		0x0
Preloader - 	SBC enabled:		False
Preloader - 	SLA enabled:		False
Preloader - 	DAA enabled:		False
Preloader - 	SWJTAG enabled:		False
Preloader - 	EPP_PARAM at 0x600 after EMMC_BOOT/SDMMC_BOOT:	False
Preloader - 	Root cert required:	False
Preloader - 	Mem read auth:		False
Preloader - 	Mem write auth:		False
Preloader - 	Cmd 0xC8 blocked:	False
Preloader - Get Target info
Preloader - 	HW subcode:		0x8a00
Preloader - 	HW Ver:			0xcb00
Preloader - 	SW Ver:			0x0
DA_handler - Device is unprotected.
DA_handler - Device is in Preloader-Mode :(
DALegacy - Uploading legacy da...
DALegacy - Uploading legacy stage 1 from MTK_AllInOne_DA_5.2136.bin
legacyext - Legacy DA2 is patched.
legacyext - Legacy DA2 CMD F0 is patched.
Preloader - Jumping to 0x200000
Preloader - Jumping to 0x200000: ok.
DALegacy - Got loader sync !
DALegacy - Reading nand info
DALegacy - Reading emmc info
DALegacy - ACK: 04029b
DALegacy - Setting stage 2 config ...
DALegacy - Uploading stage 2...
DeviceClass - USBError(19, 'No such device (it may have been disconnected)')
```

**Comment by brunothedev** [comment]
@bkerler: Is there any way i can help?

**Comment by markussbk** [comment]
I found LG-K120E, which also has MT6735M chip, in my box of electronics and I would also be glad to help out, if you need some more details.
As Bruno said, this device has no alternative OS support :(

**Comment by brunothedev** [comment]
@markussbk i am currently trying to run PostmarketOS(A linux distribution for smartphones) on it, [you can see it here](https://wiki.postmarketos.org/wiki/LG_K4_2016_(lg-k130f))

**Comment by bkerler** [comment]
Support should be possible. However some tests will be needed.

**Comment by bkerler** [comment]
If someone can send a device or let me debug it via remote session I can try to implement support.

**Comment by bkerler** [comment]
Never mind, I have a similar device at home ... support should be rolling out soon

**Comment by brunothedev** [comment]
@brunothedev Tried rn with the lattest commits, it seems to be stuck on ```Done |--------------------------------------------------| 0.0% Read (Sector 0x0 of 0x2) 0.00 MB/s
``` for some reason. But still better than crashing

**Comment by bkerler** [comment]
Ok, will have a closer look at the preloader then

**Comment by bkerler** [comment]
Please try commit [1a1f871](https://github.com/bkerler/mtkclient/commit/1a1f8719f53acddb24c9d030126683de7c022222)

**Comment by brunothedev** [comment]
@bkerler:
```Done |--------------------------------------------------| 0.0% Read (Sector 0x0 oDone |--------------------------------------------------| 0.0% Read (Sector 0x0 oDone |--------------------------------------------------| 0.0% Read (Sector 0x0 of 0x400) 0.00 MB/ssej - HACC init
sej - HACC run
sej - HACC terminate
sej - HACC init
sej
sej - [LIB]: SEJ Legacy Hardware seems not to be configured correctly. Results may be wrong.
sej - HACC run
sej - HACC terminate
Traceback (most recent call last):
  File "/usr/local/bin/mtk", line 4, in <module>
    __import__('pkg_resources').run_script('mtkclient==1.6.2', 'mtk')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 720, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1559, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python3.11/dist-packages/mtkclient-1.6.2-py3.11.egg/EGG-INFO/scripts/mtk", line 855, in <module>
    mtk = Main(args).run(parser)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/mtkclient-1.6.2-py3.11.egg/mtkclient/Library/mtk_main.py", line 633, in run
    da_handler.handle_da_cmds(mtk, cmd, self.args)
  File "/usr/local/lib/python3.11/dist-packages/mtkclient-1.6.2-py3.11.egg/mtkclient/Library/mtk_da_cmd.py", line 765, in handle_da_cmds
    v = mtk.daloader.seccfg(args.flag)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/mtkclient-1.6.2-py3.11.egg/mtkclient/Library/mtk_daloader.py", line 260, in seccfg
    return self.lft.seccfg(lockflag)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/mtkclient-1.6.2-py3.11.egg/mtkclient/Library/legacy_ext.py", line 182, in seccfg
    if not sc_org.parse(seccfg_data):
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/mtkclient-1.6.2-py3.11.egg/mtkclient/Library/seccfg.py", line 182, in parse
    self.error("Unknown V3 seccfg encryption !")
    ^^^^^^^^^^
AttributeError: 'seccfgV3' object has no attribute 'error'```. It showed this, gonna try a few more though

**Comment by bkerler** [comment]
I think I fixed this error, but it still doesn't look good. Guess I do still need that hardware to implement it correctly.

**Comment by brunothedev** [comment]
@bkerler: What hardware are you using for testing?

**Comment by bkerler** [comment]
I had a Hisense C30. Most devices I have use V4 based, or V3 SW crypted but none use V3 HW encrypted like the LG K4 as it seems.

**Comment by brunothedev** [comment]
@bkerler: This seccfg must be LG specific, i could try to contribute, do you have any documentation on seccfg reverse engineering? I could try to send you my phone too

**Comment by bkerler** [comment]
Remote session via Teamviewer might be sufficient. You can contact me via Telegram @DaBeejay.

**Comment by brunothedev** [comment]
i don't have telegram(nor a phone at all, in fact), do you have a matrix account?

**Comment by bkerler** [comment]
yep. @bkerler:gnuradio.org

**Comment by bkerler** [comment]
Solved with commit 6404991
